nautobot/utilities/templatetags/buttons.py

```diff
@@ -59,7 +62,10 @@ def edit_button(instance, use_pk=False, key="slug"):
     else:
         kwargs = {"pk": instance.pk}
 
-    url = reverse(viewname, kwargs=kwargs)
+    try:
+        url = reverse(viewname, kwargs=kwargs)
```

Comment:
Obviously in the case where you need the actual URL and not just validation the `viewname` you'll still need this code, but the duplication in calling `_get_viewname()` can be eliminated.